Failures in API Integration tests when a response is unordered

|Wazuh version|Component|
|---|---|
| Master | Wazuh API integration tests |

## Description

Sometimes a test can fail when the order of the answer is not the expected one. For instance, this `GET /agents/upgrade_result` failed because the agents in the response were `006`, `005`, `007` instead of  `005`, `006`, `007`:
```
AssertionError: assert [{'agent': '006', 'task_id': 1},\n {'agent': '005', 'task_id': 2},\n {'agent': '007', 'task_id': 3}] == [{'agent': '005',\n  'task_id': <tavern.util.loader.IntSentinel object at 0x7ff520fac280>},\n {'agent': '006',\n  'task_id': <tavern.util.loader.IntSentinel object at 0x7ff520fac1c0>},\n {'agent': '007',\n  'task_id': <tavern.util.loader.IntSentinel object at 0x7ff520fac130>}]
```

It would be necessary to investigate why `tavern` considers this incorrect. It could be related to the fact that the answer is a list, which does preserve the order, unlike dictionaries.

It should also be studied if it is an error when the socket does not return the expected order or if, on the contrary, it is an expected/acceptable behavior.

## Checks

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)
<!-- If changes are made to any of the following components, uncomment the corresponding line 
- [ ] **Integration tests** without failures for API configuration (`/wazuh-qa/tests/integration/test_api/test_config/`)
- [ ] **System tests** for agent enrollment process (`/wazuh-qa/tests/system/test_cluster/test_agent_enrollment`)
- [ ] **System tests** for agent info sync process in cluster (`/wazuh-qa/tests/system/test_cluster/test_agent_info_sync`)
- [ ] **System tests** for agent key polling (`/wazuh-qa/tests/system/test_cluster/test_agent_key_polling`)
- [ ] **System tests** for JWT invalidation (`/wazuh-qa/tests/system/test_jwt_invalidation`)
-->

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

--- a/api/test/integration/tavern_utils.py
+++ b/api/test/integration/tavern_utils.py
@@ -187,7 +187,7 @@ def test_sort_response(response, key=None, reverse=False):
 
 
 def test_validate_data_dict_field(response, fields_dict):
-    assert fields_dict, f'Fields dict is empty'
+    assert fields_dict, "Fields dict is empty"
     for field, dikt in fields_dict.items():
         field_list = response.json()['data'][field]
 

--- a/framework/wazuh/agent.py
+++ b/framework/wazuh/agent.py
@@ -983,7 +983,7 @@ def upgrade_agents(agent_list: list = None, wpk_repo: str = None, version: str =
                 else:
                     raise WazuhInternalError(error_code, cmd_error=True, extra_message=agent_result['message'])
 
-    result.affected_items.sort(key=lambda x: x['agent'])
+    result.affected_items.sort(key=operator.itemgetter('agent'))
 
     return result
 
@@ -1077,7 +1077,7 @@ def get_upgrade_result(agent_list: list = None, filters: dict = None, q: str = N
                 else:
                     raise WazuhInternalError(error_code, cmd_error=True, extra_message=task_result['message'])
 
-    result.affected_items.sort(key=lambda x: x['agent'])
+    result.affected_items.sort(key=operator.itemgetter('agent'))
 
     return result
 

--- a/framework/wazuh/agent.py
+++ b/framework/wazuh/agent.py
@@ -931,8 +931,7 @@ def upgrade_agents(agent_list: list = None, wpk_repo: str = None, version: str =
 
         # Add non active agents to failed_items
         non_active_agents = [agent['id'] for agent in data['items'] if agent['status'] != 'active']
-        [result.add_failed_item(id_=agent, error=WazuhError(1707))
-         for agent in non_active_agents]
+        [result.add_failed_item(id_=agent, error=WazuhError(1707)) for agent in non_active_agents]
         non_active_agents = set(non_active_agents)
 
         # Add non eligible agents to failed_items
@@ -948,7 +947,7 @@ def upgrade_agents(agent_list: list = None, wpk_repo: str = None, version: str =
         eligible_agents = agent_list - not_found_agents - non_active_agents - non_eligible_agents
 
         # Transform the format of the agent ids to the general format
-        eligible_agents = sorted([int(agent) for agent in eligible_agents])
+        eligible_agents = [int(agent) for agent in eligible_agents]
 
         agents_result_chunks = [eligible_agents[x:x + 500] for x in range(0, len(eligible_agents), 500)]
 
@@ -1036,8 +1035,7 @@ def get_upgrade_result(agent_list: list = None, filters: dict = None, q: str = N
 
         # Add non active agents to failed_items
         non_active_agents = [agent['id'] for agent in data['items'] if agent['status'] != 'active']
-        [result.add_failed_item(id_=agent, error=WazuhError(1707))
-         for agent in non_active_agents]
+        [result.add_failed_item(id_=agent, error=WazuhError(1707)) for agent in non_active_agents]
         non_active_agents = set(non_active_agents)
 
         # Add non eligible agents to failed_items
@@ -1053,7 +1051,7 @@ def get_upgrade_result(agent_list: list = None, filters: dict = None, q: str = N
         eligible_agents = agent_list - not_found_agents - non_active_agents - non_eligible_agents
 
         # Transform the format of the agent ids to the general format
-        eligible_agents = sorted([int(agent) for agent in eligible_agents])
+        eligible_agents = [int(agent) for agent in eligible_agents]
 
         agents_result_chunks = [eligible_agents[x:x + 500] for x in range(0, len(eligible_agents), 500)]
 

--- a/framework/wazuh/agent.py
+++ b/framework/wazuh/agent.py
@@ -984,6 +984,8 @@ def upgrade_agents(agent_list: list = None, wpk_repo: str = None, version: str =
                 else:
                     raise WazuhInternalError(error_code, cmd_error=True, extra_message=agent_result['message'])
 
+    result.affected_items.sort(key=lambda x: x['agent'])
+
     return result
 
 
@@ -1076,6 +1078,8 @@ def get_upgrade_result(agent_list: list = None, filters: dict = None, q: str = N
                 # Upgrade error for all agents, internal server error
                 else:
                     raise WazuhInternalError(error_code, cmd_error=True, extra_message=task_result['message'])
+
+    result.affected_items.sort(key=lambda x: x['agent'])
 
     return result
 

